core/node/state_keeper/src/testonly/test_batch_executor.rs

```diff
@@ -756,20 +756,19 @@ impl StateKeeperIO for TestIO {
         Ok(())
     }
 
-    async fn reject(&mut self, tx: &Transaction, error: &str) -> anyhow::Result<()> {
+    async fn reject(&mut self, tx: &Transaction, reason: UnexecutableReason) -> anyhow::Result<()> {
         let action = self.pop_next_item("reject");
         let ScenarioItem::Reject(_, expected_tx, expected_err) = action else {
             panic!("Unexpected action: {:?}", action);
         };
         assert_eq!(tx, &expected_tx, "Incorrect transaction has been rejected");
-        if let Some(expected_err) = expected_err {
-            assert!(
-                error.contains(&expected_err),
-                "Transaction was rejected with an unexpected error. Expected part was {}, but the actual error was {}",
-                expected_err,
-                error
-            );
-        }
+        assert!(
+            reason == expected_err,
+            "Transaction was rejected with an unexpected error. Expected part was {}, but the actual error was {}",
+            expected_err,
+            reason
+        );
```

Comment:
Nit: If you use `assert_eq!` macro, you wouldn't need to add the second sentence to the message, since it would show non-equal arguments by default as a part of the panic message.